### PR TITLE
Wait for auto_relay to finish in integration tests

### PR DESCRIPTION
**Pull request by ljoss17**
Closes: #XXX

## Description

Add two traits to the test abstraction:
* `CanWaitForAck`: Queries pending acknowledgments, and waits if there are any.
* `CanShutdown`:
  * The implementation for the current relayer does nothing.
  * The implementation for the relayer-next calls Tokio `JoinHandle::abort` and waits for the `auto_relay` to be finished.

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests: integration (for Hermes) or unit/mock tests (for modules).
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/relayer-components-extra/src/relay/impls/auto_relayers/parallel_bidirectional.rs [hunk 1]
```diff
@@ -39,6 +39,9 @@ where
                 .await;
         });
 
+        // Wait for handle1 and handle2 to finish.
+        // Equivalent to Join.
+        // TODO: Confirm with Soares
         handle1.into_future().await;
```
[the remark this answers is not in the document]
Comment (romac) [reply crates/relayer-components-extra/src/relay/impls/auto_relayers/parallel_bidirectional.rs:45]: In any case, it might be slightly cleaner with [`tokio::join!`](https://docs.rs/tokio/latest/tokio/macro.join.html#runtime-characteristics) which will await the two handles in parallel:

```rust
tokio::join!(handle1, handle2);
```

File: tools/integration-test/Cargo.toml [hunk 2]
```diff
@@ -21,6 +21,7 @@ ibc-relayer-components = { version = "0.1.0", path = "../../crates/relayer-compo
 ibc-relayer-runtime    = { path = "../../crates/relayer-runtime" }
 ibc-relayer-cosmos     = { path = "../../crates/relayer-cosmos" }
 ibc-test-framework = { path = "../test-framework" }
+tokio = { version = "1.0", features = ["rt"] }
```
Comment (romac) [inline tools/integration-test/Cargo.toml:24]: Let's move this one line below together with non ibc- dependencies?

File: crates/relayer-components-extra/src/relay/impls/auto_relayers/parallel_bidirectional.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ljoss17) [reply crates/relayer-components-extra/src/relay/impls/auto_relayers/parallel_bidirectional.rs:45]: I think the `into_future` is used in case a runtime other than Tokio is used

[the remark this answers is not in the document]
Comment (romac) [reply crates/relayer-components-extra/src/relay/impls/auto_relayers/parallel_bidirectional.rs:45]: Oh I didn't know we were aiming to be runtime-agnostic. Worthy goal but a bit overkill imho. Anyway, then perhaps the spawner could expose a `join` method. Although in this context, the current solution is fine as-is.

**Review by romac — APPROVED** [review]

File: crates/relayer-cosmos/src/base/impls/transaction.rs [hunk 3]
```diff
@@ -218,7 +218,8 @@ fn split_events_by_messages(in_events: Vec<AbciEvent>) -> Vec<Vec<Arc<AbciEvent>
 
     for event in in_events.into_iter() {
         if event.kind == "message"
-            && event.attributes.len() == 1
+            // TODO: What is the purpose of this filter ?
+            //&& event.attributes.len() == 1
```
Comment (romac) [inline crates/relayer-cosmos/src/base/impls/transaction.rs:222]: Why this change?

Comment (ljoss17) [reply crates/relayer-cosmos/src/base/impls/transaction.rs:247]: I added more details to the reason behind the change. Let me know if it is clear enough or if I should add more

Comment (romac) [reply crates/relayer-cosmos/src/base/impls/transaction.rs:247]: All good, thanks!
